way_to_home/tests/unittests/user_profile/testviews.py

```diff
@@ -0,0 +1,95 @@
+"""
+UserProfile view tests
+========================
+This module provides complete testing for all CustomUser's views functions.
+"""
+
+from django.test import TestCase, Client
+from unittest import mock
+from custom_user.models import CustomUser
+from user_profile.models import UserProfile
+from django.urls import reverse
+import json
+
+
+class UserProfileViewTest(TestCase):
+    """Test Case that provides testing of User Profile view"""
+    def setUp(self):
+        self.url = reverse('profile')
+
+        self.first_user = CustomUser.objects.create(
+            id=1001,
+            email='user@mail.com',
+            phone_number='+380111111111',
+            is_active=True
+        )
+        self.first_user.set_password('1111Bb')
+        self.first_user.save()
+
+        self.user_profile = UserProfile.objects.create(
+            id=1111,
+            user=self.first_user,
+            first_name='Jhon',
+            last_name='Doe'
+        )
+
+        self.second_user = CustomUser.objects.create(
+            id=2002,
+            email='user2@mail.com',
+            phone_number='+380111111112',
+        )
+        self.second_user.set_password('2222Bb')
+        self.second_user.save()
+
+        self.client = Client()
+        self.client.login(email='user@mail.com', password='1111Bb')
+
+        self.second_client = Client()
+        self.second_client.login(email='user2@mail.com', password='2222Bb')
+
+    def test_get(self):
+        expected_response = {
+            'id': 1111,
+            'first_name': 'Jhon',
+            'last_name': 'Doe',
+            'user_id': 1001
+        }
+
+        response = self.client.get(self.url)
+        self.assertJSONEqual(json.dumps(expected_response), json.loads(response.content))
+        self.assertEqual(response.status_code, 200)
+
+    def test_get_fail(self):
+        response = self.second_client.get(self.url)
+        self.assertEquals(response.status_code, 400)
```

Comment:
Are you sure that you catch this response from your view? I think it was returned from middleware. Because your second user has not `is_active=True` and he was not authorized.

Reply:
I do in fact get a 400 and not 403 response, therefore it's not from middleware

Reply:
Yes, my bad. I checked it also. :)